pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -512,8 +560,14 @@ func (s *ServiceEntryStore) edsUpdate(instances []*model.ServiceInstance) {
 			})
 	}
 
-	for k, eps := range endpoints {
-		_ = s.XdsUpdater.EDSUpdate(s.Cluster(), string(k.hostname), k.namespace, eps)
+	if push {
+		for k, eps := range endpoints {
+			s.XdsUpdater.EDSUpdate(s.Cluster(), string(k.hostname), k.namespace, eps)
```

Comment:
IMO this is a premature optimization that makes the code a lot more confusing. Do you have any evidence we actually need this? I don't see how calling 

`EDSUpdate`, `EDSUpdate`

is any slower than

`EDSCacheUpdate`, `EDSCacheUpdate`, `ConfigUpdate`

since the requests are merged anyways

Reply:
@howardjohn We can not assume all these requests are merged. As to confusion, I think this is because we were coupling cache update and push request sending together.

Reply:
We have denounce, I would be shocked if it's not merged. If we really have this concern maybe we should have a helper function that does the cache update and a push. I really don't want to pollute the core interfaces though.

Reply:
I can understand your concern, i can further investigate how to make this more clear

Reply:
The debounce is 100ms (I remember it is 10ms before) by default now , and if we have so many push requests. The latency can be very huge(10s) in theory.